pipeline_steps/common.groovy

```diff
@@ -439,24 +449,72 @@ def prepareRpcGit(String branch = "auto", String dest = "/opt"){
 
     print("Repo: ${env.RPC_REPO} Branch: ${branch}")
 
-    // checkout used instead of git as a custom refspec is required
-    // to checkout pull requests
-    checkout([$class: 'GitSCM',
-      branches: [[name: branch]],
-      doGenerateSubmoduleConfigurations: false,
-      extensions: [[$class: 'CleanCheckout']],
-      submoduleCfg: [],
-      userRemoteConfigs: [
-        [
-          url: env.RPC_REPO,
-          refspec: '+refs/pull/*:refs/remotes/origin/pr/* +refs/heads/*:refs/remotes/origin/*'
-        ]
-      ]
-    ]) // checkout
-    sh "git submodule update --init"
+    clone_with_pr_refs(env.RPC_REPO, branch)
   } // dir
 }
 
+// Clone repo with Refspecs required for PRs.
+// Should't need to supply any params to checkout a PR merged with the base.
+// Uses shell+git to avoid hostname verification failures with
```

Comment:
Sounds like https://stackoverflow.com/questions/38461705/checkout-jenkins-pipeline-git-scm-with-credentials.  The failure there was because the credentials passed in did not match the URL (ie. username/key for SSH).  I'd prefer to use checkout if at all possible!